app/src/main/java/com/hover/stax/bounty/BountyListAdapter.java

```diff
@@ -0,0 +1,109 @@
+package com.hover.stax.bounty;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.hover.stax.R;
+import com.hover.stax.actions.Action;
+import com.hover.stax.utils.Utils;
+
+import java.util.List;
+
+class BountyListAdapter extends RecyclerView.Adapter<BountyListAdapter.BountyListViewHolder> {
+	private static final String TAG = "BountyListAdapter";
+	private List<BountyAction> bountyActionList;
+	private final SelectListener selectListener;
+	private final Context context;
+
+	BountyListAdapter(List<BountyAction> actionList, SelectListener selectListener, Context context) {
+		this.bountyActionList = actionList;
+		this.selectListener = selectListener;
+		this.context = context;
+	}
+
+	@NonNull
+	@Override
+	public BountyListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+		View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.home_list_item, parent, false);
+		return new BountyListViewHolder(view);
+	}
+
+	@SuppressLint("SetTextI18n")
+	@Override
+	public void onBindViewHolder(@NonNull BountyListViewHolder holder, int position) {
+		BountyAction ba = bountyActionList.get(position);
+		Action a = ba.a;
+		updateStatusLayoutAndSetClickListener(holder, ba);
+		holder.content.setText(a.getFullDescription(context));
+		holder.amount.setText(a.getBountyAmountWithCurrency());
+	}
+
+	private void updateStatusLayoutAndSetClickListener(BountyListViewHolder holder, BountyAction ba) {
+		if (ba.a.bounty_is_open == 0) { //Bounty is closed and done
```

Comment:
This is close, but the green state only shows if this user is the one that did the bounty. So the first condition should be is_open == 0 AND last transaction != null, and then add another condition for just is_open == 0 that shows the text crossed out. (see image)

Also, when there are 2 or more conditions in an if else that are the same like this you should reduce code duplication by creating a method that takes the differences as parameters and set everything there.

![image](https://user-images.githubusercontent.com/735966/110719451-bcbb6100-81c1-11eb-9728-1ec86e65c267.png)